plugin/ra/src/main/java/org/opencds/cqf/ruler/ra/r4/RiskAdjustmentProvider.java

```diff
@@ -0,0 +1,256 @@
+package org.opencds.cqf.ruler.ra.r4;
+
+import ca.uhn.fhir.jpa.searchparam.SearchParameterMap;
+import ca.uhn.fhir.rest.annotation.Operation;
+import ca.uhn.fhir.rest.annotation.OperationParam;
+import ca.uhn.fhir.rest.param.DateParam;
+import ca.uhn.fhir.rest.param.ParamPrefixEnum;
+import ca.uhn.fhir.rest.param.ReferenceParam;
+import org.hl7.fhir.r4.model.Bundle;
+import org.hl7.fhir.r4.model.CodeableConcept;
+import org.hl7.fhir.r4.model.Coding;
+import org.hl7.fhir.r4.model.DateType;
+import org.hl7.fhir.r4.model.Extension;
+import org.hl7.fhir.r4.model.MeasureReport;
+import org.hl7.fhir.r4.model.Meta;
+import org.hl7.fhir.r4.model.Observation;
+import org.hl7.fhir.r4.model.OperationOutcome;
+import org.hl7.fhir.r4.model.Parameters;
+import org.hl7.fhir.r4.model.Quantity;
+import org.hl7.fhir.r4.model.Resource;
+import org.opencds.cqf.ruler.behavior.r4.MeasureReportUser;
+import org.opencds.cqf.ruler.provider.DaoRegistryOperationProvider;
+import org.opencds.cqf.ruler.utility.TypedBundleProvider;
+import java.util.Collections;
+import java.util.Map;
+
+public class RiskAdjustmentProvider extends DaoRegistryOperationProvider implements MeasureReportUser {
+
+	private static String suspectTypeUrl = "http://hl7.org/fhir/us/davinci-ra/StructureDefinition/ra-suspectType";
+	private static String evidenceStatusUrl = "http://hl7.org/fhir/us/davinci-ra/StructureDefinition/ra-evidenceStatus";
+	private static String evidenceStatusDateUrl = "http://hl7.org/fhir/us/davinci-ra/StructureDefinition/ra-evidenceStatusDate";
+
+	private String visited;
+
+	@Operation(name = "$risk-adjustment", idempotent = true, type = MeasureReport.class)
+	public Parameters riskAdjustment(
+		@OperationParam(name = "type", min = 1) String type,
+		@OperationParam(name = "periodStart", min = 1) String periodStart,
```

Comment:
These min parameters don't actually do what you would expect.

Care Gaps has an example of enforcing parameter requirements: https://github.com/DBCG/cqf-ruler/blob/de723e8c9504e1311579f2c1bcaaf07fdbafde71/plugin/cr/src/main/java/org/opencds/cqf/ruler/cr/r4/provider/CareGapsProvider.java#L279.

Reply:
That only works on GET requests... Perfectly valid to call this operation with a POST.